_P248_TempHumidity_AHT1x.ino

```diff
@@ -0,0 +1,339 @@
+
+
+#define AHT10_ADDRESS_0X38         0x38  //chip I2C address no.1 for AHT10/AHT15/AHT20, address pin connected to GND
+#define AHT10_ADDRESS_0X39         0x39  //chip I2C address no.2 for AHT10 only, address pin connected to Vcc
+
+#define AHT10_INIT_CMD             0xE1  //initialization command for AHT10/AHT15
+#define AHT20_INIT_CMD             0xBE  //initialization command for AHT20
+#define AHT10_START_MEASURMENT_CMD 0xAC  //start measurment command
+#define AHT10_NORMAL_CMD           0xA8  //normal cycle mode command, no info in datasheet!!!
+#define AHT10_SOFT_RESET_CMD       0xBA  //soft reset command
+
+#define AHT10_INIT_NORMAL_MODE     0x00  //enable normal mode
+#define AHT10_INIT_CYCLE_MODE      0x20  //enable cycle mode
+#define AHT10_INIT_CMD_MODE        0x40  //enable command mode
+#define AHT10_INIT_CAL_ENABLE      0x08  //load factory calibration coeff
+
+
+#define AHT10_DATA_MEASURMENT_CMD  0x33  //no info in datasheet!!! my guess it is DAC resolution, saw someone send 0x00 instead
+#define AHT10_DATA_NOP             0x00  //no info in datasheet!!!
+
+
+#define AHT10_MEASURMENT_DELAY     80    //at least 75 milliseconds
+#define AHT10_POWER_ON_DELAY       40    //at least 20..40 milliseconds
+#define AHT10_CMD_DELAY            350   //at least 300 milliseconds, no info in datasheet!!!
+#define AHT10_SOFT_RESET_DELAY     20    //less than 20 milliseconds
+
+#define AHT10_FORCE_READ_DATA      true  //force to read data
+#define AHT10_USE_READ_DATA        false //force to use data from previous read
+#define AHT10_ERROR                0xFF  //returns 255, if communication error is occurred
+
+ 
+#define AHTX0_I2CADDR_DEFAULT 0x38   ///< AHT default i2c address
+#define AHTX0_CMD_CALIBRATE 0xE1     ///< Calibration command
+#define AHTX0_CMD_TRIGGER 0xAC       ///< Trigger reading command
+#define AHTX0_CMD_SOFTRESET 0xBA     ///< Soft reset command
+#define AHTX0_STATUS_BUSY 0x80       ///< Status bit for busy
+#define AHTX0_STATUS_CALIBRATED 0x08 ///< Status bit for calibrated
+
+
+#ifdef USE_AHT2x
+  #define AHTX_CMD     0xB1 // Cmd for AHT2x
+#else
+  #define AHTX_CMD     0xE1 // Cmd for AHT1x
+#endif
+
+uint8_t AHTSetCalCmd[3]    = { AHTX_CMD, 0x08, 0x00 }; // load factory calibration coeff
+uint8_t AHTSetCycleCmd[3]  = { AHTX_CMD, 0x28, 0x00 }; // enable cycle mode
+uint8_t AHTMeasureCmd[3]   = { 0xAC, 0x33, 0x00 };     // start measurment command
+uint8_t AHTResetCmd        =   0xBA;                   // soft reset command
+
+uint8_t          AHT10_rawDataBuffer[6] = {AHT10_ERROR, 0, 0, 0, 0, 0};
+
+
+bool AHT1XStartMeasurement(uint8_t address) {
+  Wire.beginTransmission(address);
+  Wire.write(AHTMeasureCmd, 3);
+  if (Wire.endTransmission() != 0)
+    return false;
+  return true;
+}
+
+bool AHT1XReadMesurement(uint8_t address) {
+  
+  Wire.requestFrom(address, (uint8_t) 6);
+
+  for(uint8_t i = 0; Wire.available() > 0; i++) {
+     AHT10_rawDataBuffer[i] = Wire.read();
```

Comment:
You might want to add another check to avoid a buffer overrun error, if it unexpectedly returns more than 6 bytes.

Reply:
ok.